contracts/test/TestConversionHelpers.sol

```diff
@@ -0,0 +1,139 @@
+pragma solidity 0.4.24;
+
+import "./helpers/Assert.sol";
+import "../common/ConversionHelpers.sol";
+
+
+contract TestConversionHelpers {
+    uint256 constant internal FIRST = uint256(keccak256("0"));
+    uint256 constant internal SECOND = uint256(keccak256("1"));
+    uint256 constant internal THIRD = uint256(keccak256("2"));
+
+    function testUintArrayConvertedToBytes() public {
+        uint256[] memory arr = new uint256[](3);
+        arr[0] = FIRST;
+        arr[1] = SECOND;
+        arr[2] = THIRD;
+        uint256 arrLength = arr.length;
+
+        // Do conversion
+        bytes memory arrBytes = ConversionHelpers.dangerouslyCastUintArrayToBytes(arr);
+
+        // Check length
+        Assert.equal(arrBytes.length, arrLength * 32, "should have correct length as bytes array");
+
+        // Check memory position (conversion should be in place)
+        uint256 arrMemLoc;
+        uint256 arrBytesMemLoc;
+        assembly {
+            arrMemLoc := arr
+            arrBytesMemLoc := arrBytes
+        }
+        Assert.equal(arrMemLoc, arrBytesMemLoc, "should have same memory location after conversion");
+    }
```

Comment:
We could also check values here

```suggestion

        // Check values
        uint256 first;
        uint256 second;
        uint256 third;
        assembly {
            first := mload(add(arrBytesMemLoc, 0x20))
            second := mload(add(arrBytesMemLoc, 0x40))
            third := mload(add(arrBytesMemLoc, 0x60))
        }
        Assert.equal(first, FIRST, "should have correct first value");
        Assert.equal(second, SECOND, "should have correct second value");
        Assert.equal(third, THIRD, "should have correct third value");
    }
```

Reply:
Hehe yes... I was thinking about doing this and then got a bit lazy to do the mloads. It's a bit of a superfluous test since we're just accessing raw memory that we know we set, but still good to make sure the cast doesn't do something funny to the memory 👍.